caffe2/operators/cross_entropy_op.cc

```diff
@@ -522,7 +522,22 @@ in the dataset later as (true) positive example.
 )DOC")
     .NumInputs(2)
     .NumOutputs(1)
-    .IdenticalTypeAndShapeOfInputDim(0, 0)
+    .TensorInferenceFunction([](const OperatorDef& def,
+                                const vector<TensorShape>& in) {
+      CAFFE_ENFORCE_EQ(in[0].dims_size(), in[1].dims_size());
+      for (size_t i = 0; i < in[0].dims_size(); ++i) {
+        CAFFE_ENFORCE_EQ(in[0].dims(i), in[1].dims(i));
+      }
+      vector<TensorShape> out(1);
+      out[0] = in[1];
+      if (!in[0].dims().empty()) {
+        for (size_t i = 0; i + 1 < in[0].dims_size(); ++i) {
+          out[0].add_dims(in[0].dims(i));
```

Comment:
This doesn't look correct to me, shouldn't the output shape equal to `input[0].shape[:-1]`? Note you have copied the input[1]'s shape to out[0] already at line 531.